File: api-code-examples/python/author-list.py
import iroh
import asyncio


async def main():
    node = await iroh.IrohNode.memory()
    node_id = await node.node_id()
    print("Started Iroh node: {}".format(node_id))

    author = await node.author_create()
    print("Created author: {}".format(author))

    authors = await node.author_list()
    print("Authors:")
    for auth in authors:
        print("\t{}".format(auth))

# Output:
# Started Iroh node: wbwpkauwmitrwwhmw534w3u6sxyhvbivuepcdze5jd3zeqpgxyfa
# Created author: z2h3f3tozgz67g3ewvu7yxj5vtoddwoohzi2jgyrtgfl7tzocpda
# Authors:
#	fllmuelqe7lrzmlwmmydtperpafnyd6vkzukqh2l7xqg6cvjstia
#	x2ckktvt4ftjcmrb4xxmlajs7vcidkdsmxcsbd5vejezpcu6653a

asyncio.run(main())

File: api-code-examples/go/author-list.go
package main

import (
	"fmt"

	"github.com/n0-computer/iroh-ffi/iroh-go/iroh"
)

func main() {
	node, err := iroh.NewIrohNode("iroh_data_dir")
	if err != nil {
		panic(err)
	}

	author, err := node.AuthorCreate()
	if err != nil {
		panic(err)
	}
	fmt.Println("Created author:", author.ToString())

	authors, err := node.AuthorList()
	if err != nil {
		panic(err)
	}

	fmt.Println("Authors:")
	for _, author := range authors {
		fmt.Println("\t", author.ToString())
	}
}
